### PR TITLE
rustPlatform.maturinBuildHook: fix postBuild hook to use ./dist contract

### PR DESCRIPTION
**Pull request by blaggacao**
This rebuilds approx 1800 packages, hence targeting `staging`.

###### Description of changes

While maturin build hook is part of the rust tooling, it targets python packages.

Some `postBuild`, such as for example the `python-relaxe-deps-hook` depend on
the wheel being already placed inside `./dist`.

This PR ensures that contract and fixes the only package that had used the
maturin build hook and a potentially conflicting custom `postBuild` hook.

Detected via: `rg 'postBuild' $(rg  'maturinBuildHook' . -Nol)`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
technically the change in rendezvous should be an extra commit

**Comment by blaggacao** [comment]
> should be an extra commit

done via https://github.com/NixOS/nixpkgs/pull/243859/commits/a501d7e4bb8fe2bf9253758ba5e4a13b86a2758a :+1:

File: pkgs/servers/matrix-synapse/plugins/rendezvous.nix
Comment (figsoda) [inline pkgs/servers/matrix-synapse/plugins/rendezvous.nix:1]: small nit: there is a typo in the commit message, the attribute should be `matrix-synapse-plugins.matrix-http-rendezvous-synapse`

Comment (blaggacao) [reply pkgs/servers/matrix-synapse/plugins/rendezvous.nix:1]: Thanks for your review and accustomed attention to detail! :wink:  Fixed in [force-push](https://github.com/NixOS/nixpkgs/compare/a501d7e4bb8fe2bf9253758ba5e4a13b86a2758a..670a15018d8e13af208c1913b48ae250fe7c9ffe)

**Review by figsoda — APPROVED** [review]

**Pull request merged**